### PR TITLE
Expand implementation of TagsEntityInterface and TagService

### PR DESCRIPTION
**Pull request by demiankatz**
This PR makes more incremental progress on backporting tag refactoring from #2233.

### CODE REVIEW

**Comment by demiankatz** [comment]
There is still considerably more work to do with the TagService, but this is all I have time for today, and I think merging it would be useful since it's easier to resolve conflicts with #2233 in smaller chunks, and this will remove a significant number of changed files from the diff list on that PR. I'll do another round of backporting when time permits.

File: module/VuFind/src/VuFind/Db/Entity/TagsEntityInterface.php [hunk 1]
```diff
@@ -40,4 +40,26 @@
...
+     * @return int
+     */
+    public function getId(): int;
+
+    /**
+     * Tag setter
+     *
+     * @param string $tag Tag
+     *
+     * @return Tags
```
Comment (aleksip) [inline module/VuFind/src/VuFind/Db/Entity/TagsEntityInterface.php]: `TagsEntityInterface`

File: module/VuFind/src/VuFind/Db/Service/TagService.php [hunk 2]
```diff
@@ -44,6 +44,60 @@ class TagService extends AbstractDbService implements TagServiceInterface, \VuFi
...
+     * @return array
+     */
+    public function getForResource(
+        string $id,
+        string $source = DEFAULT_SEARCH_BACKEND,
+        int $limit = 0,
+        int $list = null,
+        int $user = null,
+        string $sort = 'count',
+        int $userToCheck = null
```
Comment (aleksip) [inline module/VuFind/src/VuFind/Db/Service/TagService.php]: Should the three `int`:s set to `null` be typed as `?int`? In the PHPDoc also.

File: module/VuFind/src/VuFind/Db/Service/TagServiceInterface.php [hunk 3]
```diff
@@ -42,6 +42,50 @@
...
+     * @return array
+     */
+    public function getForResource(
+        string $id,
+        string $source = DEFAULT_SEARCH_BACKEND,
+        int $limit = 0,
+        int $list = null,
+        int $user = null,
+        string $sort = 'count',
+        int $userToCheck = null
```
Comment (aleksip) [inline module/VuFind/src/VuFind/Db/Service/TagServiceInterface.php]: See comment in `TagService`

**Review by aleksip — APPROVED** [review]

**Pull request merged**